### PR TITLE
EE-1222: Express MAX_PAYMENT in Gas

### PR DESCRIPTION
**Pull request by darthsiroftardis**
Changelog:

- Changed MAX_PAYMENT to be represented as `Gas` as opposed to simply `U512`
- Accounted for above change

### CODE REVIEW

**Comment by sacherjj** [comment]
bors try

File: execution_engine/src/core/engine_state/mod.rs [hunk 1]
```diff
@@ -1173,7 +1173,8 @@ where
             Err(error) => return Ok(ExecutionResult::precondition_failure(error.into())),
         };
 
-        let max_payment_cost = Motes::new(*MAX_PAYMENT);
+        let max_payment_cost =
+            Motes::from_gas(*MAX_PAYMENT, 1u64).ok_or(Error::GasConversionOverflow)?;
```
Comment (mpapierski) [inline execution_engine/src/core/engine_state/mod.rs:1177]: We need to use supplied gas price for this conversion

```suggestion
        let max_payment_cost =
            Motes::from_gas(*MAX_PAYMENT, deploy_item.gas_price).ok_or(Error::GasConversionOverflow)?;
```

Comment (darthsiroftardis) [reply execution_engine/src/core/engine_state/mod.rs]: Applied suggestion

**Review by henrytill — CHANGES_REQUESTED** [review]
Holding until further notice

**Review by EdHastingsCasperLabs — APPROVED** [review]

**Comment by mpapierski** [comment]
Closed in favor of #1818 - we'll go with `u64` representation for gas limit and `Gas` itself

**Pull request closed without merging**